lib/utils/addr.go

```diff
@@ -342,3 +343,12 @@ func guessHostIP(addrs []net.Addr) (ip net.IP) {
 	}
 	return ip
 }
+
+// ReplaceUnspecifiedHost  replaces unspecified "0.0.0.0" host localhost since 0.0.0.0 is never a valid
+// principal (auth server explicitly removes it when issuing host certs) and when a reverse tunnel client used
+// establishes SSH reverse tunnel connection the host is validated against
+// the valid principal list.
+func ReplaceUnspecifiedHost(addr *NetAddr, defaultPort int) string {
```

Comment:
I think this function needs to either check for "unspecified" host itself (and return `addr` unmodified otherwise), or be renamed to "ReplaceHostWithLocalhost" or something.

Because right now it just replaces any `addr` host with localhost regardless of whether it's unspecified or not, which may cause confusion with a caller.

Reply:
👍 I have moved to `ReplaceUnspecifiedHost`